src/olympia/files/models.py

```diff
@@ -182,8 +183,18 @@ def from_upload(cls, upload, version, parsed_data=None):
             file_.save()  # This also saves the file to the filesystem.
 
         permissions = list(parsed_data.get('permissions', []))
+        if file_.manifest_version == 3:
```

Comment:
It should likely be:

```suggestion
        if file_.manifest_version > 2:
```